sonar-db/src/main/java/org/sonar/db/component/ComponentDao.java

```diff
@@ -105,31 +102,16 @@ public boolean existsById(Long id, DbSession session) {
     return mapper(session).selectEnabledFilesFromProject(rootComponentUuid);
   }
 
-  public List<ComponentDto> selectByIds(final DbSession session, Collection<Long> ids) {
-    return DatabaseUtils.executeLargeInputs(ids, new Function<List<Long>, List<ComponentDto>>() {
-      @Override
-      public List<ComponentDto> apply(@Nonnull List<Long> partition) {
-        return mapper(session).selectByIds(partition);
-      }
-    });
+  public List<ComponentDto> selectByIds(DbSession session, Collection<Long> ids) {
+    return executeLargeInputs(ids, partition -> mapper(session).selectByIds(partition));
   }
 
-  public List<ComponentDto> selectByUuids(final DbSession session, Collection<String> uuids) {
-    return DatabaseUtils.executeLargeInputs(uuids, new Function<List<String>, List<ComponentDto>>() {
-      @Override
-      public List<ComponentDto> apply(@Nonnull List<String> partition) {
-        return mapper(session).selectByUuids(partition);
-      }
-    });
+  public List<ComponentDto> selectByUuids(DbSession session, Collection<String> uuids) {
+    return executeLargeInputs(uuids, partition -> mapper(session).selectByUuids(partition));
   }
 
-  public List<String> selectExistingUuids(final DbSession session, Collection<String> uuids) {
-    return DatabaseUtils.executeLargeInputs(uuids, new Function<List<String>, List<String>>() {
-      @Override
-      public List<String> apply(@Nonnull List<String> partition) {
-        return mapper(session).selectExistingUuids(partition);
-      }
-    });
+  public List<String> selectExistingUuids(DbSession session, Collection<String> uuids) {
+    return executeLargeInputs(uuids, partition -> mapper(session).selectExistingUuids(partition));
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Replace this lambda with a method reference. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1612)